src/rgw/rgw_rest_role.cc

```diff
@@ -56,6 +57,52 @@ int RGWRestRole::verify_permission(optional_yield y)
   return 0;
 }
 
+int RGWRestRole::parse_tags()
+{
+  vector<string> keys, vals;
+  auto val_map = s->info.args.get_params();
+  const regex pattern_key("Tags.member.([0-9]+).Key");
```

Comment:
sorry to be dense, what does the "." match here?  i.e., is it matching any character in this position?

Reply:
The tags are supplied as: Tags.member.N, where N is a number. So '.' matches the dot after member.

Reply:
AWS documentation here: https://docs.aws.amazon.com/IAM/latest/APIReference/API_CreateRole.html